charts/everest/charts/common/templates/_upgrade_checks.yaml.tpl

```diff
@@ -0,0 +1,38 @@
+#
+# @param .namespace             The namespace where the operator is installed
+# @param .version               Version to upgrade to
+# @param .versionMetadataURL    The URL of the version metadata service
+#
+{{- define "everest.preUpgradeChecks" }}
+{{- $hookName := printf "everest-helm-pre-upgrade-hook" }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ $hookName }}-{{ randNumeric 6 }}
+  namespace: {{ .namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    spec:
+      containers:
+        - image: alpine:3.20
+          name: {{ $hookName }}
+          command:
+            - /bin/sh
+            - -c
+            - |
+              OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+              ARCH=$(uname -m)
+              VERSION={{ .version }}
+              apk add --no-cache --quiet curl
+              curl -sSL -o everestctl https://github.com/percona/everest/releases/download/v${VERSION}/everestctl-${OS}-${ARCH}
+              chmod -R 777 ./everestctl
+              
+              echo "Checking requirements for upgrade to version ${VERSION}"
+              ./everestctl upgrade --dry-run --version-metadata-url={{ .versionMetadataURL }}
```

Comment:
This command won't work just yet because the upgrade CLI does not create an in-cluster config. I am working on a PR to add that.

Reply:
https://github.com/percona/everest/pull/800 is merged